### PR TITLE
Fix/canvas reparent orphans

### PR DESCRIPTION
**Pull request by balazsbajorics**
**Problem:**
We want to be able to reparent elements directly to the canvas. These elements become the children of the Storyboard element, without existing inside a component. These are called Orphans. We had a half-baked support for Orphans for the last year, and we made some significant regressions in the last six months, and most recently, last week.

**Fix:**
Allow Orphans to be reparented to - from the Canvas root. Selecting them is already supported on master.

**Commit Details:**
- Added data-utopia-valid-paths to the Storyboard, containing orphan paths – this allows the dom-based select mode hooks to find the elements
- The reparent target lookup now uses the DOM, which meant that now we need to feed the _transient_ canvas state's selected views to the reparent functions. (because the transient canvas state is rendered to the DOM, that's what the dom lookup will see, that's what we need to filter / validate against)

### CODE REVIEW

File: editor/src/core/shared/template-path.ts [hunk 1]
```diff
@@ -149,6 +149,17 @@ export function toString(target: TemplatePath): string {
   }
 }
 
+export function toStringOptional(target: TemplatePath | null | undefined): string | null {
```
Comment (balazsbajorics) [inline editor/src/core/shared/template-path.ts:152]: unfortunately I couldn't extend toString to be able to return an optional null without having to change too much irrelevant code – so I made a second variant of the function instead

Comment (seanparsons) [reply editor/src/core/shared/template-path.ts:152]: I'd argue for using `optionalMap` with the regular `toString` function personally, but this doesn't appear to be being used?

**Review by seanparsons — APPROVED** [review]

File: editor/src/core/shared/template-path.ts (hunk 1 above)
Comment (balazsbajorics) [reply editor/src/core/shared/template-path.ts:152]: I meant to write that I wrote this for debugging convenience, not used in the code. I didn't think about optionalMap, I guess I can just remove this then!

Comment (balazsbajorics) [reply editor/src/core/shared/template-path.ts]: removed!

**Review by enidemi — APPROVED** [review]

**Pull request merged**